pkg/auth/auth_handler.go

```diff
@@ -232,6 +244,17 @@ func (a *AuthHandler) CreateAuthenticator() func(name, in string, authenticate s
 				}
 				return true, nil, common.NewInfraError(http.StatusUnauthorized, err)
 			}
+			//this code is part of the authorization process and should move to authz_handler
+			//after https://github.com/go-openapi/runtime/issues/158 is resolved
+			if payload, ok := p.(*ocm.AuthPayload); ok && !payload.IsAdmin {
```

Comment:
I think all this should move to a function and be called from here.
If you create a module with this, new function and the isClusterOwnedByUser() function in it, Will also be easier to UT. 
This way you will probably not need to have postgres in the auth_test since this function can be mocked

and the function could be checked better when it is in another module

Reply:
resolved off line